lib/fluent/config/configure_proxy.rb

```diff
@@ -208,9 +208,21 @@ def overwrite_defaults(other) # other is owner plugin's corresponding proxy
         end
       end
 
-      def option_value_type!(name, opts, key, klass)
-        if opts.has_key?(key) && !opts[key].is_a?(klass)
-          raise ArgumentError, "#{name}: #{key} must be a #{klass}, but #{opts[key].class}"
+      def option_value_type!(name, opts, key, klass=nil, type: nil)
+        if opts.has_key?(key)
+          if klass && !opts[key].is_a?(klass)
+            raise ArgumentError, "#{name}: #{key} must be a #{klass}, but #{opts[key].class}"
+          end
+          case type
+          when :boolean
```

Comment:
`bool` is better for consistency?
`config_param` uses `:bool` for boolean type parameter.